### DISCUSSION TITLE
Details of “Simplify” task.

### DISCUSSION THREAD

**Discussion by zhao-w-en** [deepmodeling/dpgen#642] [Q&A]
Hello developers,
Recently I read about the **transfer learning work** you mentioned on Zhihu and looked up related papers. It is a useful feature as we won't spent long time on retraining a new dp model when we chose an inappropriate functional in last training process. Although the JSON file has been presented on Zhihu, I noticed that it seems to mention some new keywords. Unfortunately, I don't see much about this on dpGen's Github page. 
So **the first question** is, **where should I look up the relevant keywords** so that I can write a JSON file that suits my project.
**Another issue** is that I have some questions about the transfer learning process in **YiNan Wang et al.** 's paper. Why should we exclude the data with model deviations **smaller** than 0.05 eV Å−1. Is this to reduce redundant data?
`The data with model deviations smaller than 0.05 eV Å−1 are excluded, and then at most 100 data points are randomly picked
from the remaining dataset (those data with model deviations larger than 0.05 eV Å−1)`
The following provides a part of the code of the JSON file on Zhihu, which seems not completely consistent with the description in the paper. And does the setting of energy mean that it is not used as a filter for data?
```   
    "labeled": false,
    "init_pick_number":100,
    "iter_pick_number":100,
    "e_trust_lo":1e10,
    "e_trust_hi":1e10,
    "f_trust_lo":0.20,
    "f_trust_hi":100.00, 
```
Looking forward to your reply.
zhaow
[https://www.zhihu.com/org/shen-du-shi-neng-deep-potential](url)

**Comment by Carrotkingdom** [comment]
Hi W-En,

For the first question, all the relevant keywords are presented on Zhihu. Those not mentioned are normal keywords of Deepmd-kit, so I didn't explain them. Which keywords don't you understand? You can list it here and I will explain it. Later we will update the keywords of transfer learning in Github.

For the second question, we excluded those data because the deviation is rather small. We believed that including these data will not improve the model evidently. This idea is similar to the common model-deviation process.   
As for the JSON file you provided here, the f_trust_lo is not the same as I used in the paper. That's OK, you may test which is the best for your system. The energy deviation is not the filter, so it is set to be very large.

Let me know if you have any other questions,

Yinan

**Comment by zhao-w-en** [comment]
Hello nan,
Thank you very much for your reply.  

I have a few questions on keywords like` "training_iter0_model"`.  I have developed a model by **dp train** (compress) **without using dpgen**. Will it matter if I use a compressed model and set` "numb_models"` to 1?  

Beyond that, in `"default_training_param" `part, you set` "trainable"` to `false`(str) for embedding net , and set` "tranable"` to `false,false,false,true`(list) for fitting net. Does it means `"trainable"`  can accept two types? And I also wonder why there are four options in a list as the fitting net only have three hidden layers.

In `"loss"` part I noticed that `"start_pref_e"` is equal to `"limit_pref_e"`, `"start_pref_f"` is equal to `"limit_pref_f"`，only `"start_pref_v"` and `"limit_pref_v"` changed. I don't quite understand the rationale for setting parameters like this. This setting makes the net seem to fit energy more than atomic forces

In `fp` part,  I didn't find explanation of `"fp_skip_bad box"` in github. And I want to know how to get rid of configurations where atoms are too close together, as you mentioned in Zhihu. For keywords such as `"init_pick_number"` and `"iter_pick_number"`, it appears to be related to the setting of `"fp_task_max"`.   Do they have to be set the same?

I am sorry that I have many questions, and I would appreciate your reply.
zhaowen

**Reply by Carrotkingdom** [reply]
I have a few questions on keywords like "training_iter0_model". I have developed a model by dp train (compress) without using dpgen. Will it matter if I use a compressed model and set "numb_models" to 1?
----DPGEN doesn’t support training compressed models now, but kit does. I suggest preparing more than 1 model with different seeds. If you want to use DPGEN to do transfer learning, please do not start from the compressed model.
Beyond that, in "default_training_param" part, you set "trainable" to false(str) for embedding net , and set "trainable" to false,false,false,true(list) for fitting net. Does it means "trainable" can accept two types? 
----Yes, of course. 
And I also wonder why there are four options in a list as the fitting net only has three hidden layers.
----https://docs.deepmodeling.org/projects/deepmd/en/master/_modules/deepmd/fit/ener.html?highlight=trainable
See this website for the explanation.
In "loss" part I noticed that "start_pref_e" is equal to "limit_pref_e", "start_pref_f" is equal to "limit_pref_f"，only "start_pref_v" and "limit_pref_v" changed. I don't quite understand the rationale for setting parameters like this. This setting makes the net seem to fit energy more than atomic forces
----In most cases, we do the init-model to get a better description of energy. You can refer to https://mp.weixin.qq.com/s/cy5ApBFrqYakCS3fAkNIKA for a better understanding of the settings parameter.  
In fp part, I didn't find an explanation of "fp_skip_bad box" in github. 
----It means to skip those structures of ‘bad box’, like the ratio of the y-axis length and x-axis length is larger than 5: length_ratio:5; the ratio of the z-axis length and x-axis length is larger than 5: height_ratio:5. 
And I want to know how to get rid of configurations where atoms are too close together, as you mentioned in Zhihu. 
----This part has not been merged into the DPGEN now. You can achieve it with your code. 
For keywords such as "init_pick_number" and "iter_pick_number", it appears to be related to the setting of "fp_task_max". Do they have to be set the same?
----"init_pick_number" and "iter_pick_number" are the numbers of structures you picked from the dataset generated by DP-GEN.  "fp_task_max" is the number of relabeled data in each iteration during the transfer learning. They are not related and can be different.
